bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -845,6 +854,30 @@ protected void release(boolean wait)
       }
     }
 
+    private boolean canRelease(boolean writer)
```

Comment:
'release' is a two step process. It includes decrementing reference count and if resulting reference count is zero and spooling is enabled, deciding based on eviction policy whether the block should be also spooled to disk. The first part (decrementing reference count) is done always. `canRelease` determines whether the block that is released can be also spooled to disk.